### PR TITLE
[TR] PIM-4265: fix behat scenario

### PR DESCRIPTION
**Pull request by sumbobyboys**

### CODE REVIEW

File: features/Context/DataGridContext.php [hunk 1]
```diff
@@ -790,7 +790,12 @@ public function iPressMassDeleteButton()
      */
     public function iPressSequentialEditButton()
     {
-        $this->getCurrentPage()->sequentialEdit();
+        $this->getMainContext()->spin(function () {
+            $this->getCurrentPage()->sequentialEdit();
+
```
Comment (jmleroux) [inline features/Context/DataGridContext.php]: sequentialEdit method already use a spin method, so i think it's overkill. :)

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -279,8 +279,13 @@ public function changePageSize($num)
...
         $element = $this->getElement('Grid toolbar')->find('css', '.page-size');
-        $element->find('css', 'button')->click();
-        $element->find('css', sprintf('ul.dropdown-menu li a:contains("%d")', $num))->click();
+
+        $this->spin(function () use ($element, $num) {
+            $element->find('css', 'button')->click();
+            $element->find('css', sprintf('ul.dropdown-menu li a:contains("%d")', $num))->click();
+
+            return true;
+        });
```
Comment (jmleroux) [inline features/Context/Page/Base/Grid.php]: maybe separate in two spin methods

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/simple-select-field.js [hunk 3]
```diff
@@ -90,8 +90,8 @@ define(
                     )
                 ).promise();
             },
-            updateModel: function (event) {
-                var data = event.currentTarget.value;
+            updateModel: function () {
+                var data = this.$('input[type="hidden"].select-field').get(0).value;
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/simple-select-field.js]: Could we not use `[type="hidden"].select-field:first` to be consistent with the event

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/text-field.js [hunk 4]
```diff
@@ -22,13 +22,13 @@ define([
...
-                'change input': 'updateModel'
+                'change input[type="text"]:first': 'updateModel'
             },
             renderInput: function (context) {
                 return this.fieldTemplate(context);
             },
-            updateModel: function (event) {
-                var data = event.currentTarget.value;
+            updateModel: function () {
+                var data = this.$('input[type="text"]').get(0).value;
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/text-field.js]: same here

**Comment by jmleroux** [comment]
GTM otherwise

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/price-collection-field.js [hunk 5]
```diff
@@ -26,9 +26,9 @@ define([
 
             return this.fieldTemplate(context);
         },
-        updateModel: function (event) {
+        updateModel: function () {
             var data = [];
-            var $elements = $(event.currentTarget).parents('.price-collection-field').find('.price-input');
+            var $elements = this.$('.price-collection-field:first').find('.price-input');
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/price-collection-field.js]: `this.$('.price-collection-field .price-input');` works too I think.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/simple-select-field.js (hunk 3 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/simple-select-field.js]: we want the dom element not the jquery one. So same result but not the same way to handle it :)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/panel/history.js [hunk 6]
```diff
@@ -73,6 +73,7 @@ define(
 
                     this.delegateEvents();
                     this.renderExtensions();
+                    this.getParent().resize();
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/panel/history.js]: No way to call it in an event in the parent ? : /

**Comment by phaseinducer** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/panel/history.js (hunk 6 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/panel/history.js]: Yeah, I will try to work on that today :)

**Pull request merged**